warp10/src/main/java/io/warp10/script/functions/STORE.java

```diff
@@ -56,19 +56,22 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
     }
     
     if (count > 0) {
-      for (int i = count - 1; i >= 0; i--) {
+      Object[] objectsToStore = stack.popn(count);
```

Comment:
Use get(n) in the loop then a dropn instead of calling popn which will allocate a new array.